mobile/cmp/multiZoneGrid/MultiZoneGridModel.ts

```diff
@@ -0,0 +1,421 @@
+    limits: Partial<Record<Zone, ZoneLimit>>;
+    delimiter: string;
+    restoreDefaultsFn: () => Awaitable<boolean>;
+    restoreDefaultsWarning: ReactNode;
+
+    private _defaultState; // initial state provided to ctor - powers restoreDefaults().
+    @managed persistenceModel: MultiZonePersistenceModel;
+
+    constructor(config: MultiZoneGridConfig) {
+        super();
+        makeObservable(this);
+
+        const {
+            columns,
+            limits,
+            mappings,
+            leftColumnSpec,
+            rightColumnSpec,
+            delimiter,
+            multiZoneMapperModel,
+            restoreDefaultsFn,
+            restoreDefaultsWarning = MultiZoneGridModel.DEFAULT_RESTORE_DEFAULTS_WARNING,
+            persistWith,
+            ...rest
+        } = config;
+
+        this.availableColumns = columns.map(it => ({...it, hidden: true}));
+        this.limits = limits;
+        this.mappings = this.parseMappings(mappings);
+
+        this.leftColumnSpec = leftColumnSpec;
+        this.rightColumnSpec = rightColumnSpec;
+        this.delimiter = delimiter ?? ' • ';
+        this.restoreDefaultsFn = restoreDefaultsFn;
+        this.restoreDefaultsWarning = restoreDefaultsWarning;
+
+        this._defaultState = {
+            mappings: this.mappings,
+            sortBy: rest.sortBy,
+            groupBy: rest.groupBy
+        };
+
+        this.gridModel = this.createGridModel(rest);
+        this.mapperModel = this.parseMapperModel(multiZoneMapperModel);
+        this.persistenceModel = persistWith
+            ? new MultiZonePersistenceModel(this, persistWith)
+            : null;
+
+        this.addReaction({
+            track: () => [this.leftColumnSpec, this.rightColumnSpec],
+            run: () => this.gridModel.setColumns(this.getColumns())
+        });
+    }
+
+    /**
+     * Restore the mapping, sorting, and grouping configs as specified by the application at
+     * construction time. This is the state without any user changes applied.
+     * This method will clear the persistent grid state saved for this grid, if any.
+     *
+     * @returns true if defaults were restored
+     */
+    async restoreDefaultsAsync(): Promise<boolean> {
+        if (this.restoreDefaultsWarning) {
+            const confirmed = await XH.confirm({
+                title: 'Please Confirm',
+                icon: Icon.warning(),
+                message: this.restoreDefaultsWarning,
+                confirmProps: {
+                    text: 'Yes, restore defaults',
+                    intent: 'primary'
+                }
+            });
+            if (!confirmed) return false;
+        }
+
+        const {mappings, sortBy, groupBy} = this._defaultState;
+        this.setMappings(mappings);
+        this.setSortBy(sortBy);
+        this.setGroupBy(groupBy);
+
+        this.persistenceModel?.clear();
+
+        if (this.restoreDefaultsFn) {
+            await this.restoreDefaultsFn();
```

Comment:
use conditional eval on 183 to match line  180?

Reply:
I tried that, but IJ complained about it with the `await`